ksql-serde/src/main/java/io/confluent/ksql/serde/tls/ThreadLocalCloseable.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.serde.tls;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public abstract class ThreadLocalCloseable<T extends Closeable> implements Closeable {
```

Comment:
Could we use composition rather than inheritance here? There is nothing really abstract about this class and composition is usually a nicer way of doing it, particularly if it is with an interface...

Reply:
Agreed. Will fix.